### PR TITLE
Fix build failure on Jetpack 4.6

### PR DESCRIPTION
**Pull request by CoderHam**
**Description**: 

Modify commit id of nlohmann/json to unblock jetpack 4.6 build for ORT 1.8.1

**Motivation and Context**
- Why is this change required? What problem does it solve?

I was attempting a build with a specific commit (d14b08d09cee43b42fa9ac361a78d1cbdbceddef) off master that enables TRT 8 support in ORT 1.8.1. This specific compilation issue is seen on a Jetson Xavier

```
In file included from /triton/onnxruntime/onnxruntime/test/providers/kernel_def_hash_test.cc:65:0:
/triton/onnxruntime/cmake/external/json/single_include/nlohmann/json.hpp: In member function ‘void nlohmann::detail::serializer<BasicJsonType>::dump_escaped(const string_t&, bool) [with BasicJsonType = nlohmann::basic_json<>]’:
/triton/onnxruntime/cmake/external/json/single_include/nlohmann/json.hpp:15817:10: error: ‘%.2X’ directive output may be truncated writing between 2 and 8 bytes into a region of size 3 [-Werror=format-truncation=]
     void dump_escaped(const string_t& s, const bool ensure_ascii)
          ^~~~~~~~~~~~
/triton/onnxruntime/cmake/external/json/single_include/nlohmann/json.hpp:15817:10: note: directive argument in the range [0, 2147483647]
In file included from /usr/include/stdio.h:862:0
```

- Fixes https://github.com/nlohmann/json/issues/2572

This is required by Triton Inference Server for supporting the Onnxruntime backend on Jetson.
cc @askhade

### CODE REVIEW

File: cgmanifests/submodules/cgmanifest.json [hunk 1]
```diff
@@ -255,7 +255,7 @@
       "component": {
         "type": "git",
         "git": {
-          "commitHash": "db78ac1d7716f56fc9f1b030b715f872f93964e4",
```
Comment (jywu-msft) [inline cgmanifests/submodules/cgmanifest.json:258]: this is just the manifest. but did you update the submodule itself?
i'm in the process of setting up a device with jetpack 4.6 to repro and validate.

Comment (CoderHam) [reply cgmanifests/submodules/cgmanifest.json:258]: Updated the submodule as well

**Comment by snnn** [comment]
Let's hold on this. The new one is not released yet. @faxu and @pranavsharma  may have concerns.

**Comment by jywu-msft** [comment]
I made a PR for silencing the build warning, if we don't want to update the submodule to unreleased version.
https://github.com/microsoft/onnxruntime/pull/8701

**Pull request closed without merging**